sno/working_copy/base.py

```diff
@@ -221,6 +221,17 @@ def assert_db_tree_match(self, tree, *, table_name="*"):
             raise Mismatch(wc_tree_id, expected_tree_id)
         return wc_tree_id
 
+    def tracking_changes_count(self, dataset=None):
+        with self.session() as db:
+            if dataset is not None:
+                r = db.execute(
+                    f"SELECT COUNT(*) FROM {self.TRACKING_TABLE} WHERE table_name=?;",
+                    (dataset.table_name),
+                )
+            else:
+                r = db.execute(f"SELECT COUNT(*) FROM {self.TRACKING_TABLE}")
+            return r.fetchone()[0]
```

Comment:
```suggestion
            return r.scalar()
```

Reply:
Changed a bunch of these in files in this PR... will do a follow up PR where I fix this everywhere.